project/ScalazBuild.scala

```diff
@@ -4,13 +4,12 @@ import Keys._
 object Scalaz extends Build {
   val testDeps = Seq("org.scalacheck" %% "scalacheck" % "1.13.4" % "test")
 
-  def module(prjName: String) = Project(
-    id = prjName,
-    base = file(prjName)).settings(
+  def stdSettings(prjName: String) = Seq(
     name := s"scalaz-$prjName",
-    scalaVersion := "2.11.11",
+    scalaVersion := "2.12.2",
     scalacOptions ++= Seq("-feature","-deprecation", "-Xlint", "-language:higherKinds",
-                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8"),
+                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8", "-opt:l:project",
```

Comment:
I find suspicious that `-opt:l:project` is required here, what is the impact on performance?

Inlining is known to have cause issues in the best in users library/projects.